docs/Validation-and-Serialization.md

```diff
@@ -11,234 +11,187 @@ Fastify uses a schema-based approach, and even if it is not mandatory we recomme
-  + `myField: { $ref: 'http://url.com/sh.json#/definitions/foo'}` will search for a shared schema added with `$id: 'http://url.com/sh.json'` and will use the field `definitions.foo`
-  + `myField: { $ref: 'http://url.com/sh.json#foo'}` will search for a shared schema added with `$id: 'http://url.com/sh.json'` and it will look inside of it for object with `$id: '#foo'`
++ `myField: { $ref: '#foo'}` will search for field with `$id: '#foo'` inside the current schema
++ `myField: { $ref: '#/definitions/foo'}` will search for field `definitions.foo` inside the current schema
++ `myField: { $ref: 'http://url.com/sh.json#'}` will search for a shared schema added with `$id: 'http://url.com/sh.json'`
++ `myField: { $ref: 'http://url.com/sh.json#/definitions/foo'}` will search for a shared schema added with `$id: 'http://url.com/sh.json'` and will use the field `definitions.foo`
++ `myField: { $ref: 'http://url.com/sh.json#foo'}` will search for a shared schema added with `$id: 'http://url.com/sh.json'` and it will look inside of it for object with `$id: '#foo'`
 
 
-More examples:
-
-**`$ref-way`** usage examples:
+**Simple usage:**
 
 ```js
 fastify.addSchema({
-  $id: 'http://example.com/common.json',
+  $id: 'http://example.com/',
   type: 'object',
   properties: {
     hello: { type: 'string' }
   }
 })
 
-fastify.route({
-  method: 'POST',
-  url: '/',
+fastify.post('/', {
+  handler () {},
   schema: {
     body: {
       type: 'array',
-      items: { $ref: 'http://example.com/common.json#/properties/hello' }
+      items: { $ref: 'http://example.com#/properties/hello' }
     }
-  },
-  handler: () => {}
+  }
 })
 ```
 
-**`replace-way`** usage examples:
+**`$ref` as root reference:**
 
 ```js
-const fastify = require('fastify')()
-
 fastify.addSchema({
-  $id: 'greetings',
+  $id: 'commonSchema',
   type: 'object',
   properties: {
     hello: { type: 'string' }
   }
 })
 
-fastify.route({
-  method: 'POST',
-  url: '/',
+fastify.post('/', {
+  handler () {},
   schema: {
-    body: 'greetings#'
-  },
-  handler: () => {}
+    body: { $ref: 'commonSchema#' },
+    headers: { $ref: 'commonSchema#' }
+  }
 })
+```
 
-fastify.register((instance, opts, done) => {
-
-  /**
-   * In children's scope can use schemas defined in upper scope like 'greetings'.
-   * Parent scope can't use the children schemas.
-   */
-  instance.addSchema({
-    $id: 'framework',
-    type: 'object',
-    properties: {
-      fastest: { type: 'string' },
-      hi: 'greetings#'
-    }
-  })
-
-  instance.route({
-    method: 'POST',
-    url: '/sub',
-    schema: {
-      body: 'framework#'
-    },
-    handler: () => {}
-  })
+<a name="get-shared-schema"></a>
+#### Retrieving the shared schemas
 
-  done()
-})
-```
+If the validator and the serializer will be customized, the `.addSchema` will not be useful since the actors are no more
+under the Fastify controls.
+So, to access the schemas added to the Fastify's instance, you can simply use `.getSchemas()`:
```

Comment:
```suggestion
So, to access the schemas added to the Fastify instance, you can simply use `.getSchemas()`:
```

Is this true? The preceding text states that `addSchema` doesn't work. So how would `getSchemas` work?